tests/specs/infinite-gas.k

```diff
@@ -0,0 +1,55 @@
+requires "evm.md"
+
+module INFINITE-GAS
+    imports EVM
+
+
+ // Symbolic Gas
+
+    syntax Int ::= #gas ( Int , Int , Int )  [function]  // startGas, nonMemory, memory
+ //-----------------------------------------------------------------------------------
+
+ //
+ // Infinite Gas
+ // Here we use the construct `#gas` to count _up_ gas used instead of counting down.
+ // This allows (i) computing final gas used, and (ii) never stopping because of out-of-gas.
+ //
+
+    syntax Int ::= #gas ( Int ) [function, functional, no-evaluators]
+ // -----------------------------------------------------------------
+    rule #gas(_, _, _) => #gas(0)
+
+    rule #gas(G) -Int G' => #gas(G +Int G') [simplification]
+
+    rule #gas(_)  <Int I => false requires I <=Int pow256 [simplification]
+    rule #gas(_) <=Int I => false requires I <=Int pow256 [simplification]
+    rule #gas(_) >=Int I => true  requires I <=Int pow256 [simplification]
+    rule I <=Int #gas(_) => true  requires I <=Int pow256 [simplification]
```

Comment:
@ehildenb Can we mark these rules `symbolic` to ensure that the `concrete` rules below are always preferred?

Reply:
Hmmmm, I'm not opposed, but I guess I'd be surprised it if led to a performance improvement. These are all heuristics, so whichever is the first to apply is good enough.

Up to you though, if you have reason to believe it will improve performance somehow then it makes sense.